h2o-core/src/main/java/water/util/GetLogsFromNode.java

```diff
@@ -76,26 +83,26 @@ public void doIt() {
     }
 
     //here is the code for the method
-    private void zipDir(String dir2zip, String pathInArchive, ByteArrayOutputStream baos, ZipOutputStream zos) throws IOException
-    {
+    private void archiveDir(String dir, String pathInArchive, ByteArrayOutputStream baos, LogArchiveWriter writer) {
       try
       {
         //convert paths represented as strings into File instances
-        File sourceDir = new File(dir2zip);
+        File sourceDir = new File(dir);
         File destinationDir = new File(pathInArchive);
         //get a listing of the directory content
         String[] dirList = sourceDir.list();
+        if (dirList == null)
+          return;
         byte[] readBuffer = new byte[4096]; 
-        int bytesIn = 0;
-        //loop through dirList, and zip the files
+        //loop through dirList, and archive the files
         for(int i=0; i<dirList.length; i++)
         {
           File sourceFile = new File(sourceDir, dirList[i]);
           File destinationFile = new File(destinationDir, dirList[i]);
           if(sourceFile.isDirectory())
           {
             //if the File object is a directory, call this
```

Comment:
Can we remove `//if the File object is a directory, call this` and `//loop again` comments? They don't seem to bring any additional value in this case

Reply:
There are actually more places where we could remove unnecessary comments, probably not worth it mixing this with this PR